/**:
  ros__parameters:
    # Robot information
    robot_info:
      # Name of the robot
      name: "sobit_light"
      # Version of the robot
      version: "1.0.0"
      # Description of the robot
      morphology:
        # Type of the robot (e.g., "mobile_manipulator", "humanoid", "quadruped", etc.)
        type: "mobile_manipulator"
        parts:
          - head
          - manipulator
          - end_effector
          - mobile_base
        head:
          joint_names:
            - "head_yaw_joint"
            - "head_pitch_joint"
        manipulator:
          joint_names:
            - "arm_shoulder_roll_joint"
            - "arm_shoulder_pitch_joint"
            - "arm_shoulder_pitch_sub_joint"
            - "arm_elbow_pitch_joint"
            - "arm_forearm_roll_joint"
            - "arm_wrist_pitch_joint"
            - "arm_wrist_roll_joint"
        end_effector:
          joint_names:
            - "hand_joint"
        mobile_base:
          joint_names:
            - "base_l_drive_wheel_joint"
            - "base_r_drive_wheel_jointt"
      # List of sensors on the robot
      sensors:
        types:
          - rgbd # color and depth camera
          - lidar # light detection and ranging sensor
          - fish_eye # fisheye camera
        rgbd:
          names: ["head_camera", "hand_camera"]
          models: ["realsense_d435", "realsense_d455"]
        lidar:
          names: ["lidar_front"]
          models: ["unk"]
        fish_eye:
          names: ["camera_front", "camera_rear"]
          models: ["unk", "unk"]

    # User information for contact
    user_info:
      # Name of the user
      name: ""
      # Location of the user
      location: ""
      # Contact email for the user
      email: ""

    # ROS bag recording configuration
    rosbag_config:
      # Path to the directory where the recorded bag files will be saved
      record_directory: "~/vla_ros_bags"
      # List of topics to record (cameras, joint_states, tf, sensors, etc.)
      topics_to_record:
        - /sobit_light/back_camera/camera_info
        - /sobit_light/back_camera/image_raw
        - /sobit_light/back_camera/image_raw/camera_info
        - /sobit_light/back_camera/image_raw/compressed
        - /sobit_light/front_camera/camera_info
        - /sobit_light/front_camera/image_raw
        - /sobit_light/front_camera/image_raw/camera_info
        - /sobit_light/front_camera/image_raw/compressed
        - /sobit_light/hand_camera/color/camera_info
        - /sobit_light/hand_camera/color/image_rect_raw
        - /sobit_light/hand_camera/color/image_rect_raw/compressed
        - /sobit_light/hand_camera/color/image_rect_raw/compressedDepth
        - /sobit_light/hand_camera/color/image_rect_raw/theora
        - /sobit_light/hand_camera/color/metadata
        - /sobit_light/hand_camera/depth/camera_info
        - /sobit_light/hand_camera/depth/image_rect_raw
        - /sobit_light/hand_camera/depth/image_rect_raw/compressed
        - /sobit_light/hand_camera/depth/image_rect_raw/compressedDepth
        - /sobit_light/hand_camera/depth/image_rect_raw/theora
        - /sobit_light/hand_camera/depth/metadata
        - /sobit_light/hand_camera/extrinsics/depth_to_color
        - /sobit_light/hand_camera/extrinsics/depth_to_depth
        - /sobit_light/head_camera/color/camera_info
        - /sobit_light/head_camera/color/image_raw
        - /sobit_light/head_camera/color/image_raw/compressed
        - /sobit_light/head_camera/color/image_raw/compressedDepth
        - /sobit_light/head_camera/color/image_raw/theora
        - /sobit_light/head_camera/color/metadata
        - /sobit_light/head_camera/depth/camera_info
        - /sobit_light/head_camera/depth/image_rect_raw
        - /sobit_light/head_camera/depth/image_rect_raw/compressed
        - /sobit_light/head_camera/depth/image_rect_raw/compressedDepth
        - /sobit_light/head_camera/depth/image_rect_raw/theora
        - /sobit_light/head_camera/depth/metadata
        - /sobit_light/head_camera/extrinsics/depth_to_color
        - /sobit_light/imu/imu
        - /sobit_light/tof_camera/camera_info
        - /sobit_light/tof_camera/image_raw
        - /sobit_light/tof_camera/image_raw/camera_info
        - /sobit_light/tof_camera/image_raw/compressedDepth
        - /sobit_light/velocity_controller/commands
        - /sobit_light/velocity_controller/transition_event
        - /sobit_light/wheel_odometry/wheel_odometry
        - /sobit_light/lidar/scan
        - /sobit_light/manual_control/cmd_vel
        - /sobit_light/odometry/odometry
        - /sobit_light/joy
        - /sobit_light/joint_state_broadcaster/transition_event
        - /sobit_light/joint_states
        - /sobit_light/joint_trajectory_controller/controller_state
        - /sobit_light/joint_trajectory_controller/joint_trajectory
        - /sobit_light/joint_trajectory_controller/state
        - /sobit_light/joint_trajectory_controller/transition_event
        - /sobit_light/diff_controller/cmd_vel_unstamped
        - /sobit_light/diff_controller/odom
        - /sobit_light/dynamic_joint_states
        - /tf
        - /tf_static
      # List of services to record (TODO: not available until ROS 2 Jazzy)
      services_to_record:
        - ""
      # List of actions to record (TODO: not available until ROS 2 Kilted)
      actions_to_record:
        - ""
      # Format for the recorded bag files (e.g., "mcap", "sqlite3")
      conversion_format: "mcap"
      # Specify the compression mode ("none", "file", "message")
      compression_mode: "none"
      # If compression is enabled, specify the compression format ("zstd" is the only supported compression format)
      compression_format: "zstd"

    # Control settings for the gamepad
    gamepad_config:
      # Gamepad type (e.g., "dualshock4", "keyboard")
      name: "dualshock4"
      # Mapping of gamepad buttons to actions
      dualshock4:
        button_mapping:
          record: 7 # Top
          save:   6 # Left
          delete: 6 # Right
      keyboard:
        button_mapping:
          record: "r"
          save: "s"
          delete: "d"
